lib/online_order.rb

```diff
@@ -0,0 +1,106 @@
+require_relative '../lib/order'
+
+require 'csv'
+require 'awesome_print'
+
+module Grocery
+  class  OnlineOrder < Grocery::Order
+    attr_reader :customerid, :status
+
+    #overwrites initialize method for Order
+    def initialize(id, products, customerid, status)
+      super(id, products)
+      @customerid = customerid
+      @status = status
+    end
+
+    #returns a list of OnlineOrder instances where the value of the customer's ID matches the passed parameter.
+    def self.find_by_customer(customer_id)
+      online_orders_csv = CSV.read("support/online_orders.csv", 'r',headers: true).to_a
+
+      if customer_id > online_orders_csv.length
+        return nil
+      end
+
+      if customer_id >=1
+        specific_online_order = online_orders_csv [customer_id-1]
+      elsif customer_id < 0
+        specific_online_order = online_orders_csv[customer_id]
```

Comment:
I'm not sure what this code is doing. You should be looping through the list of orders looking for ones that have a matching customer ID. Something like:

```ruby
def self.find_by_customer(customer_id)
  matches = []
  all.each do |order|
    if order.customer_id == customer_id
      matches << order
    end
  end
  return matches
end
```

Or, if you want to get fancy with enumerables:

```ruby
def self.find_by_customer(customer_id)
  return all.select { |order| order.customer_id == customer_id }
end
```